guides/source/security.md

```diff
@@ -29,6 +30,147 @@ The threats against web applications include user account hijacking, bypass of a
+
+To use this feature in your application, you can run `rails generate authentication`. Here are all of the files the generator modifies and new files it adds:
+
+```bash
+$ rails generate authentication
+      invoke  erb
+      create    app/views/passwords/new.html.erb
+      create    app/views/passwords/edit.html.erb
+      create    app/views/sessions/new.html.erb
+      create  app/models/session.rb
+      create  app/models/user.rb
+      create  app/models/current.rb
+      create  app/controllers/sessions_controller.rb
+      create  app/controllers/concerns/authentication.rb
+      create  app/controllers/passwords_controller.rb
+      create  app/mailers/passwords_mailer.rb
+      create  app/views/passwords_mailer/reset.html.erb
+      create  app/views/passwords_mailer/reset.text.erb
+      create  test/mailers/previews/passwords_mailer_preview.rb
+        gsub  app/controllers/application_controller.rb
+       route  resources :passwords, param: :token
+       route  resource :session
+        gsub  Gemfile
+      bundle  install --quiet
+    generate  migration CreateUsers email_address:string!:uniq password_digest:string! --force
+       rails  generate migration CreateUsers email_address:string!:uniq password_digest:string! --force
+      invoke  active_record
+      create    db/migrate/20241010215312_create_users.rb
+    generate  migration CreateSessions user:references ip_address:string user_agent:string --force
+       rails  generate migration CreateSessions user:references ip_address:string user_agent:string --force
+      invoke  active_record
+      create    db/migrate/20241010215314_create_sessions.rb
+```
+
+As shown above, the authentication generator modifies the `Gemfile` to add the `bcrypt` gem. The generator uses the `bcrypt` gem for storing a hash of the password in the database (instead of plain-text passwords). As this process is not reversible, there's no way to go from the hash back to the password.
+
+The generator adds two migrations for creating `user` and `session` tables. Next step is to run the migrations:
+
+```bash
+$ bin/rails db:migrate
+```
+
+Then, if you visit `/session/new` in your browser (you will see this route has been added in `routes.rb`), you'll see a form that accepts an email and a password with "sign in" button. This form routes to the `SessionsController` which was added by the generator. If you provide an email/password for a user that exists in the database, you will be able to successfully authenticate with those credentials and login to the application.
+
+NOTE: After running the Authentication generator, you do need to implement your own *sign up flow* and add the necessary views, routes, and controller actions. There is no code generated that creates new `user` records and allows users to "sign up" in the first place. This is something you'll need to wire up based on the requirements of your application.
```

Comment:
Do we have anywhere in the guides where we can point readers where this is already documented?

Reply:
I don't think so.

Reply:
Didn't think so. Thanks!